cartographer/mapping_2d/local_trajectory_builder.cc

```diff
@@ -228,15 +228,18 @@ LocalTrajectoryBuilder::AddHorizontalRangeData(
     return nullptr;
   }
 
+  // indices of submaps_ where new scans are inserted are decided here...
```

Comment:
iirc the fact I wanted to point out, was that `InsertRangeData` might lead to another `submap` being created. This leads to the fact that inside `InsertRangeData` the insertion indices are not necessarily "the last" and "the one before the last". At some point that was surprising for me. 
I can still rephrase it if you prefer.